src/components/Page/components/Header/Header.tsx

```diff
@@ -116,18 +122,13 @@ export function Header({
     />
   );
 
-  const primary =
-    primaryAction &&
-    (primaryAction.primary === undefined ? true : primaryAction.primary);
+  function getPrimaryActionMarkup() {
```

Comment:
Nested function definitions tend to feel odd to me - you need to go check for any closed over variables that appear in the scope. Can this be pulled out to the top level?